src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -384,20 +390,34 @@ case class IndexLogEntry(
 
   @JsonIgnore
   lazy val allSourceFileInfos: Set[FileInfo] = {
-    relations
-      .flatMap(_.data.properties.content.fileInfos)
-      .toSet
+    relations.head.data.properties.content.fileInfos
   }
 
-  def deletedFiles: Seq[String] = {
-    relations.head.data.properties.deletedFiles
+  def deletedFiles: Set[FileInfo] = {
+    if (relations.head.data.properties.deletedFiles.isDefined) {
+      relations.head.data.properties.deletedFiles.get.fileInfos
+    } else {
+      Set()
+    }
   }
 
-  def appendedFiles: Seq[String] = {
-    relations.head.data.properties.appendedFiles
+  def appendedFiles: Set[FileInfo] = {
```

Comment:
Can you add the following comments to `def`s that return `FileInfo` with full path so we can revisit this later?
```
// FileInfo's 'name' contains the full path to the file.
```